ledger/shelley.go

```diff
@@ -441,6 +441,40 @@ type ShelleyProtocolParameterUpdate struct {
 	MinUtxoValue uint `cbor:"15,keyasint"`
 }
 
+const (
+	NonceType0 = 0
+	NonceType1 = 1
+)
+
+type Nonce struct {
+	cbor.StructAsArray
+	Type  uint `cbor:"0,keyasint"`
+	Value any  `cbor:"1,keyasint,omitempty"`
```

Comment:
Remove these `cbor` tags. They don't do anything when using `cbor.StructAsArray`. The `keyasint` and `omitempty` tags only work when dealing with a map (this is a list).